app/static/admin/js/pages/gen_form/form-donation.js

```diff
@@ -1,174 +1,239 @@
-            "</td>\n" +
-            '    <td class="field-pick_up">' +
-            donation.pick_up +
-            "</td>\n" +
-            '    <td class="field-verified">' +
-            (donation.verified
-              ? '<img src="/static/admin/img/icon-yes.svg" alt=true>'
-              : '<img src="/static/admin/img/icon-no.svg" alt=false>') +
-            "    </td>\n" +
-            "</tr>";
+            donation = data[ix];
+            store[donation.tax_receipt_no] = donation;
+            html +=
+                '<tr class="row' +
+                (ix % 2 ? 2 : 1) +
+                '" id="' +
+                donation.tax_receipt_no +
+                '" >\n' +
+                '    <td class="field-tax_receipt_no">' +
+                donation.tax_receipt_no +
+                "</td>\n" +
+                '    <td class="field-donate_date nowrap">' +
+                donation.donate_date +
+                "</td>\n" +
+                '    <td class="field-pick_up">' +
+                donation.pick_up +
+                "</td>\n" +
+                '    <td class="field-verified">' +
+                (donation.verified
+                    ? '<img src="/static/admin/img/icon-yes.svg" alt=true>'
+                    : '<img src="/static/admin/img/icon-no.svg" alt=false>') +
+                "    </td>\n" +
+                "</tr>";
         }
 
-        setDonationForm.call(this, null);
-        // scrollTo(donation_result_div);
-        donation_table_body.innerHTML = html;
-      };
-    })();
-
-    var saveDonation = function() {
-      console.log(this.serialize());
-      $.ajax({
-        beforeSend: util.csrf,
-        url: "/api/donation",
-        type: "POST",
-        dataType: "json",
-        data: {
-          donor_id: 127
-        },
-        success: function(response) {
-          console.log("Response", response);
-        },
-        error: function() {
-          console.error(arguments);
-        }
-      });
-    }.bind(this.dom.form);
-
-    $(this.dom.table.tbody).on("click", "tr", function(e) {
-      var tr = this.children;
-      var data = {};
-
-      data.tax_receipt_no = tr[0].innerText;
-      data.donate_date = tr[1].innerText;
-      data.pick_up = tr[2].innerText;
-      data.verified = tr[3].getElementsByTagName("img")[0].alt;
-
-      setDonationForm(e, data);
-      item.getItems.call(this, data.tax_receipt_no);
-      util.scrollTo(this);
-    });
-    $(this.dom.button.addNew).on("click", setDonationForm);
-    $(this.dom.button.cancel).on("click", setDonationForm);
-    $(this.dom.button.save).on("click", saveDonation);
-    $(this.dom.button.update).on("click", function() {});
-
-    return {
-      printDonationList: printDonationList
+        clearDonationForm();
+        dom.table.tbody.innerHTML = html;
     };
-  }.bind({})
+
+        var saveDonation = function () {
+
+            if (!store.donor_id) alert("Save donor first");
+
+            $.ajax({
```

Comment:
Consider creating a function that takes a callback where it takes the HTTP request type, model type, and success callback function. It will make your code a lot more readable and scalable.

Reply:
I personally prefer to have the exact function rather than having to go through all if-cases but I will look into implementing your suggestion